airflow/providers/dbt/cloud/hooks/dbt.py

```diff
@@ -0,0 +1,488 @@
+            instead of those configured in dbt Cloud.
+        :param schema_override: Optional. Override the destination schema in the configured target for this
+            job.
+        :param additional_run_config: Optional. Any additional parameters that should be included in the API
+            request when triggering the job.
+        :return: The request response.
+        """
+        if additional_run_config is None:
+            additional_run_config = {}
+
+        payload = {
+            "cause": cause,
+            "steps_override": steps_override,
+            "schema_override": schema_override,
+        }
+        payload.update(additional_run_config)
+
+        return self._run_and_get_response(
+            method="POST",
+            endpoint=f"{account_id}/jobs/{job_id}/run/",
+            payload=json.dumps(payload),
+        )
+
+    @fallback_to_default_account
+    def list_job_runs(
+        self,
+        account_id: Optional[int] = None,
+        include_related: Optional[List[str]] = None,
+        job_definition_id: Optional[int] = None,
+        order_by: Optional[str] = None,
+    ) -> List[Response]:
+        """
+        Retrieves metadata for all of the dbt Cloud job runs for an account. If a ``job_definition_id`` is
+        supplied, only metadata for runs of that specific job are pulled.
+
+        :param account_id: Optional. The ID of a dbt Cloud account.
+        :param include_related: Optional. List of related fields to pull with the run.
+            Valid values are "trigger", "job", "repository", and "environment".
+        :param job_definition_id: Optional. The dbt Cloud job ID to retrieve run metadata.
+        :param order_by: Optional. Field to order the result by. Use '-' to indicate reverse order.
+            For example, to use reverse order by the run ID use ``order_by=-id``.
+        :return: List of request responses.
+        """
+        return self._run_and_get_response(
+            endpoint=f"{account_id}/runs/",
+            payload={
+                "include_related": include_related,
+                "job_definition_id": job_definition_id,
+                "order_by": order_by,
+            },
+            paginate=True,
+        )
+
+    @fallback_to_default_account
+    def get_job_run(
+        self, run_id: int, account_id: Optional[int] = None, include_related: Optional[List[str]] = None
+    ) -> Response:
+        """
+        Retrieves metadata for a specific run of a dbt Cloud job.
+
+        :param run_id: The ID of a dbt Cloud job run.
+        :param account_id: Optional. The ID of a dbt Cloud account.
+        :param include_related: Optional. List of related fields to pull with the run.
+            Valid values are "trigger", "job", "repository", and "environment".
+        :return: The request response.
+        """
+        return self._run_and_get_response(
+            endpoint=f"{account_id}/runs/{run_id}/",
```

Comment:
Should we log an error if account_id is not passed, looks like it might construct the url with None.

Reply:
This should be caught by the `@fallback_to_default_account` decorator which wraps every hook method that calls a dbt Cloud API endpoint directly. The decorator checks to see if the `account_id` is not passed or if it is, if the value is `None`. If either of those conditions are met, the `account_id` set in the Airflow connection is used. In the event there is no `account_id` configured in the connection, this exception is raised: `raise AirflowException("Could not determine the dbt Cloud account.")`. Does that suffice?